Sources/Workspace/Workspace.swift

```diff
@@ -1845,37 +1876,62 @@ extension Workspace {
             return result
         }
 
-        func computePackages() throws -> (required: Set<PackageReference>, missing: Set<PackageReference>) {
-            let manifestsMap: [PackageIdentity: Manifest] = try Dictionary(throwingUniqueKeysWithValues:
-                self.root.packages.map { ($0.key, $0.value.manifest) } +
-                self.dependencies.map { ($0.dependency.packageRef.identity, $0.manifest) }
+        private static func computeDependencies(
+            root: PackageGraphRoot,
+            dependencies: [(manifest: Manifest, dependency: ManagedDependency, productFilter: ProductFilter, fileSystem: FileSystem)],
```

Comment:
nit: would it be better to create a struct for this instead of using a tuple?

Reply:
perhaps. this PR does not introduce this tho. we can revisit in a separate PR